File: src/main/java/my/study/repository/design/pattern/observe/java/ObserveImpl.java
package my.study.repository.design.pattern.observe.java;

/**
 * @author Djh on 2018/7/27 14:50
 * E-Mail ：1544579459@qq.com
 */
public class ObserveImpl implements Observe {
    private UseDemo.DataSet mDataSet;
    private String mObserveName;

    ObserveImpl(UseDemo.DataSet dataSet, String observeName) {
        mObserveName = observeName;
        mDataSet = dataSet;
    }

    @Override
    public void update() {
        System.out.print(mObserveName + "更新了数据: ");
        for (Integer integer : mDataSet.integerList) {
            System.out.print(integer + " ");
        }
        System.out.println(" ");
    }
}

File: src/main/java/my/study/repository/design/pattern/observe/kotlin/ObserveImpl.kt
package my.study.repository.design.pattern.observe.kotlin

/**
 * @author djh on  2018/10/10 17:00
 * @E-Mail 1544579459@qq.com
 */
class ObserveImpl(
        private val list: MutableList<Int>
) : Observe {
    
    override fun dataUpdate() {
        println("Data set is change, the last element is ${list.last()}")
    }
}